File: greedy/55.go
func canJump(nums []int) bool {
    var max int 
    for i := 0; i < len(nums); i++ {
        if i > max {
            return false
        }
        if nums[i] + i > max {
            max = nums[i] + i
        }
    }
    return true
}

File: greedy/55.java
class Solution {
    public boolean canJump(int[] nums) {
        int max = 0;
        for (int i = 0; i < nums.length; i++) {
            if (max < i) {
                return false;
            }
            if (i + nums[i] > max) {
                max = i + nums[i];
            }
        }
        return true;
    }
}
